esphome/components/rc522/__init__.py

```diff
@@ -24,6 +24,7 @@
                 cv.GenerateID(CONF_TRIGGER_ID): cv.declare_id(RC522Trigger),
             }
         ),
+        cv.Optional(CONF_GAIN, default=4): cv.Range(min=0, max=7),
```

Comment:
Don't want people putting decimals
```suggestion
        cv.Optional(CONF_GAIN, default=4): cv.int_range(min=0, max=7),
```